Modules/Loadable/Markups/Widgets/qMRMLMarkupsCurveSettingsWidget.cxx

```diff
@@ -318,6 +318,7 @@ void qMRMLMarkupsCurveSettingsWidget::onApplyCurveResamplingPushButtonClicked()
     {
     return;
     }
+  bool isClosed = (vtkMRMLMarkupsClosedCurveNode::SafeDownCast(inputNode) != nullptr);
```

Comment:
We usually use `markupsNode->GetCurveClosed()` to decide if it is a closed curve. It is because curve management infrastructure is a feature of the vtkMRMLMarkupsNode base class. There can be custom markups that take advantage of this infrastructure (and display closed curve) but derive directly from this vtkMRMLMarkupsNode base class and not from vtkMRMLMarkupsClosedCurveNode.

```suggestion
  bool isClosed = inputNode->GetCurveClosed();
```